src/Hydrator/ModelHydrator.php

```diff
@@ -13,30 +13,23 @@
  */
 final class ModelHydrator implements Hydrator
 {
+    private $arrayHydrator;
+    private $creatableFromArray;
+
+    public function __construct(ArrayHydrator $arrayHydrator, CreatableFromArray $creatableFromArray)
```

Comment:
What's the difference with passing $class? Don't you need other implementations of CreatableFromArray anyway?